azext_iot/common/shared.py

```diff
@@ -304,5 +309,33 @@ class SHAHashVersions(Enum):
     """
     Supported SHA types for generating the certificate thumbprint.
     """
+
     SHA1 = 1
     SHA256 = 256
+
+
+# Utility classes for nested edge config file values and device arguments
+class EdgeContainerAuth(NamedTuple):
```

Comment:
Is a EdgeContainerAuth valuable if no attributes are set? Typing shows nothing is required (everything optional).

Reply:
Sorry that was a bit of laziness left over from some object prototyping, I *believe* all of these are required, but there's a chance that user/pass may be optional - will investigate and update.